velox/exec/tests/SharedArbitratorTest.cpp

```diff
@@ -1839,132 +1861,22 @@ TEST_F(SharedArbitrationTest, reclaimFromCompletedJoinBuilder) {
-            return;
-          }
-          // Make sure each hash build operator has reserved memory to avoid
-          // trigger memory arbitration in test.
-          if (static_cast<MemoryPoolImpl*>(op->pool())
-                  ->testingMinReservationBytes() == 0) {
-            return;
-          }
-          // Check all the hash build operators' memory usage instead of
-          // individual operator.
-          if (op->pool()->parent()->currentBytes() < joinMemoryUsage) {
-            return;
-          }
-          if (++injectCount > numDrivers) {
-            return;
-          }
-          auto future = builderBarrier.wait();
-          if (future.wait().value()) {
-            fakeAllocationWaitFlag = false;
-            fakeAllocationWait.notifyAll();
-          }
-          // Wait for pause to be triggered.
-          taskPauseWait.await([&]() { return !taskPauseWaitFlag.load(); });
-        })));
-
-    // This indicates that a spill has been initiated by the arbitrator.
-    SCOPED_TESTVALUE_SET(
-        "facebook::velox::exec::Task::requestPauseLocked",
-        std::function<void(Task*)>([&](Task* /*unused*/) {
-          taskPauseWaitFlag = false;
-          taskPauseWait.notifyAll();
-        }));
-
-    // joinQueryCtx and fakeMemoryQueryCtx may be the same and thus share the
-    // same underlying QueryConfig.  We apply the changes here instead of using
-    // the AssertQueryBuilder to avoid a potential race condition caused by
-    // writing the config in the join thread, and reading it in the memThread.
-    std::unordered_map<std::string, std::string> config{
-        {core::QueryConfig::kSpillEnabled, "true"},
-        {core::QueryConfig::kJoinSpillEnabled, "true"},
-        {core::QueryConfig::kJoinSpillPartitionBits, "2"}};
-    joinQueryCtx->testingOverrideConfigUnsafe(std::move(config));
-
-    std::thread joinThread([&]() {
-      const auto result = runHashJoinTask(
-          vectors, joinQueryCtx, numDrivers, true, expectedResult);
-      auto taskStats = exec::toPlanStats(result.task->taskStats());
-      auto& planStats = taskStats.at(result.planNodeId);
-      ASSERT_GT(planStats.spilledBytes, 0);
-    });
-
-    std::thread memThread([&]() {
-      auto task =
-          AssertQueryBuilder(duckDbQueryRunner_)
-              .queryCtx(fakeMemoryQueryCtx)
-              .plan(PlanBuilder()
-                        .values(vectors)
-                        .addNode([&](std::string id, core::PlanNodePtr input) {
-                          return std::make_shared<FakeMemoryNode>(id, input);
-                        })
-                        .planNode())
-              .assertResults("SELECT * FROM tmp");
-    });
-    joinThread.join();
-    memThread.join();
-    waitForAllTasksToBeDeleted();
-    ASSERT_GT(arbitrator_->stats().numRequests, 0);
-  }
+  // Create a query ctx with a small capacity to trigger spilling.
+  std::shared_ptr<core::QueryCtx> queryCtx = newQueryCtx(128 << 20);
```

Comment:
just to clarify, does this mean that any attempt to spill would go via the arbitrator?

Reply:
We have threshold based spilling trigger but we don't leverage this in shared arbitrator test. In shared arbitrator test, the spilling is always triggered from arbitrator.